src/OzuCms/OzuCollectionConfig.php

```diff
@@ -31,6 +33,18 @@ public function setHasPublicationState(bool $hasState = true): self
         return $this;
     }
 
+    /**
+     * Declare which date field will trigger auto-deploy when reached
+     * @param string|null $field
+     * @return $this
+     */
+    public function setHasAutoDeployDateField(?string $fieldKey = null): self
```

Comment:
I think it should be `setAutoDeployDateField(string $fieldKey): self` (without Has and without nullable arg)
And maybe we can add an `unsetAutoDeployDateField(): self` if we need to allow to nullify this field